### DISCUSSION TITLE
How to use this display?

### DISCUSSION THREAD

**Discussion by daywink** [lovyan03/LovyanGFX#345] [Q&A] [answered]
Hello, I am unsure how to use this display, as the RD pin is missing. I have an ESP32 board and this display:

![IMG_0851-min](https://user-images.githubusercontent.com/65610123/216656261-03a909b3-4a5e-4cc4-9ec1-cf477c200f0c.png)
![IMG_0852-min](https://user-images.githubusercontent.com/65610123/216656272-03740b3f-dde6-45f1-8315-77117981fa70.png)

Can I be given an example, as well as the pinout for it to see if it works? Thanks.

* This question is for a friend, I'm posting on his behalf.

**Comment by tobozo** [comment, marked as answer]
yes it is supported, it can do 16 bits parallel data mode, only S2 or a S3 dev board have enough pins for this mode though.

it can probably do other data modes (e.g. SPI3) but you'll have to check the documentation to find that out, selecting a different mode may require a physical or software configuration depending on the model.

![image](https://user-images.githubusercontent.com/1893754/216684776-f249dda6-4c31-41eb-8b30-a6cf5a602b75.png)

See  [this folder](https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx_user/) with a few examples,

**Reply by daywink** [reply]
What about the missing RD pin? And how do I configure the different modes?

**Reply by tobozo** [reply]
maybe the RD pin is not missing and just has a different name (e.g. CS given its location on the header) ?

also if the technical specs only list one data mode for the TFT, then you can't configure a different mode

**Reply by daywink** [reply]
It doesn't make sense. There is a display CS, an SD CS and a flash CS. There is no such thing as a read pin or similar on the display.
Here is a pinout diagram, maybe you can make some sense of it?

![l](https://user-images.githubusercontent.com/65610123/216707093-76f465e6-7315-46f9-9827-12f3a7c8c7bc.jpg)

**Reply by daywink** [reply]
The datasheet:
[MAR3513-User Manua.pdf](https://github.com/lovyan03/LovyanGFX/files/10582865/MAR3513-User.Manua.pdf)

**Reply by daywink** [reply]
Ok, I got it fixed. But now there is a different issue. I will open a new discussion on it .

**Reply by tobozo** [reply]
glad you made sense of it

**Reply by lovyan03** [reply]
If a panel with a 16-bit data bus is operated with an 8-bit connection, it will not operate properly. Keep in mind that it will not be drawn in at least the correct color.

**Reply by bedirhandertli4** [reply]
@daywink  Hi! I have the exact same MAR3513/HUGA 3.5" 480x320 display and I'm trying to use it with an ESP32-S3 through LovyanGFX 16-bit parallel.

My display stays completely white. I tried both ILI9486 and ILI9488 configurations.

You mentioned here that you got it fixed. Could you please share what you changed in the wiring or configuration? It would really help me.

**Comment by TangerineDreamer** [comment]
i have a 7inch SSD1963 with esp32,i only use one range of DB0...DB7,don't use the rest of DB's
maybe you can look at this user setup #50 at : 
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup50_SSD1963_Parallel.h

**Reply by lovyan03** [reply]
Since SSD1963 can be changed by command to operate in 8-bit or 16-bit mode, it can be used in 8-bit mode.
Note that the board targeted in this discussion does not; it must be wired for all 16 bits to be drawn in the correct color.
